src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/DebugServer.java

```diff
@@ -28,6 +28,7 @@
 import sun.jvm.hotspot.runtime.*;
 import sun.jvm.hotspot.oops.*;
 
+@Deprecated(since="24", forRemoval=true)
```

Comment:
I didn't even realize this class existed. So this an (undocumented) way to start the debug server. Is there some other class that connects to it other than using `jhsdb --connect`?

Reply:
Right, I think this class is an orphan, it has no users.
The HotSpotAgent constructor mentions DebugServer in one comment, but does not use it, and probably means to say "startServer".  This comment should be updated when startServer is eventually removed.

Checking older releases, there used to be shell scripts which invoked DebugServer, and an SADebugServer class which invoked DebugServer.